code/scripting/api/objs/modelinstance.cpp

```diff
@@ -0,0 +1,323 @@
+//
+//
+
+#include "model.h"
+#include "modelinstance.h"
+#include "object.h"
+#include "vecmath.h"
+
+namespace scripting {
+namespace api {
+
+ADE_OBJ(l_ModelInstance, modelinstance_h, "modelinstance", "Model instance handle");
+
+modelinstance_h::modelinstance_h(int pmi_id)
+{
+	_pmi = model_get_instance(pmi_id);
+}
+modelinstance_h::modelinstance_h(polymodel_instance *pmi)
+	: _pmi(pmi)
+{}
+modelinstance_h::modelinstance_h()
+	: _pmi(nullptr)
+{}
+polymodel_instance *modelinstance_h::Get()
+{
+	return _pmi;
+}
+bool modelinstance_h::IsValid()
+{
+	return (_pmi != nullptr);
+}
+
+
+ADE_OBJ(l_SubmodelInstance, submodelinstance_h, "submodelinstance", "Submodel instance handle");
```

Comment:
as to stay consistent with naming, I'd suggest snake case here for the user-facing type name: submodel_instance

Reply:
ok